scripts/publish-docs.sh

```diff
@@ -71,6 +71,8 @@ git worktree add gh-pages gh-pages
 touch gh-pages/.nojekyll
 \cp -r $DOCS_FOLDER/* gh-pages
 echo "Copy javadoc to gh-pages/servicetalk/SNAPSHOT"
+# Avoid accumulating old javadocs for classes that have been moved, renamed or deleted.
+rm -rf gh-pages/servicetalk/SNAPSHOT/javadoc
```

Comment:
Great catch! 👍 
For the cases when we want to update the release version as well (for example, when we release a patch version like 0.21.1 and want to update 0.21 javadoc) we need to do the same for `gh-pages/servicetalk/$version/javadoc` below, inside the `if` statement.